### PR TITLE
[plugin.switchback@nexus] 1.0.0

### PR DESCRIPTION
**Pull request by bossanova808**
### Add-on details:

- **General**
  - Add-on name: Switchback
  - Add-on ID: plugin.switchback
  - Version number: 1.0.0
  - Kodi/repository version: nexus

- **Code location**
  - URL: https://github.com/bossanova808/plugin.switchback/
  

Kodi utility for fast switching between recently played media.

Keeps a list of recently played media - supports Kodi library episodes, movies, PVR channels, songs and non-library files.  Does not (yet?) support add-ons or PVR recordings.

The primary intended use is to make for super easy Switchback between two in progress videos (by binding a remote key to Switchback, see Wiki/Forum thread for info).

Consider this scenario:
- You are watching 'video A' with your partner.  You are interrupted and your partner needs to tend to the kids/howl at the moon.
- You navigate to 'video B' and watch some of that.
- Your partner comes back.
- You hit your one button 'Switchback' and 'video A' starts playing again, no need for tedious navigation etc.
- You are interrupted again - the moon is so very bright tonight - hit your 'Switchback' to resume 'video B' immediately, again with one magic button and no tedious navigation.
        

### Description of changes:

            v1.0.0 Initial release
        

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by bossanova808** [comment]
@basrieter As requested, now a PR to Nexus branch.

File: plugin.switchback/resources/lib/playback.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    duration:float = None
+    artist:str = None
+    album:str = None
+    tracknumber:int = None
+    channelname: str = None
+    channelnumberlabel:str = None
+    channelgroup:str = None
+
+    def toJson(self):
+        return json.dumps(self, default=lambda o: o.__dict__)
```
Comment (basrieter) [inline plugin.switchback/resources/lib/playback.py]: I would just return a dictionary here and do the dumping to string in the place where you save stuff.

File: plugin.switchback/resources/lib/store.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        Logger.info(Store.switchback_list)
+
+    @staticmethod
+    def save_switchback_list():
+        """
+        Save the Switchback list to file in JSON format
+        :return:
+        """
+        with open(Store.switchback_list_file, 'w', encoding='utf-8') as f:
+            temp_json = []
```
Comment (basrieter) [inline plugin.switchback/resources/lib/store.py]: Use my previous comment and let `toJson` return a dictionary and then just do a `json.dumps` of the full array here.

File: plugin.switchback/resources/lib/switchback_service.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    while not Store.kodi_event_monitor.abortRequested():
+        # Abort was requested while waiting. We should exit
+        if Store.kodi_event_monitor.waitForAbort(1):
+            break
+        # Otherwise, if we're playing something, record where we are up to, for later resumes
+        # (Playback record is created onAVStarted in player.py, so check here that it is available)
+        elif Store.current_playback and Store.kodi_player.isPlaying():
+            Store.current_playback.resumetime = Store.kodi_player.getTime()
+            xbmc.sleep(500)
```
Comment (basrieter) [inline plugin.switchback/resources/lib/switchback_service.py]: Use `Store.kodi_event_monitor.waitForAbort` here.

**Review by basrieter — CHANGES_REQUESTED** [review]
Please have a look at my comments.

**Comment by bossanova808** [comment]
I think that is all addressed.  

(I'm not really super clear on the `waitForAbort` vs `sleep` behaviour but let me know if this isn't right!)

**Review by basrieter — APPROVED** [review]

**Comment by basrieter** [comment]
Thanks for you contribution!

**Pull request merged**